includes/class-wcsatt-order.php

```diff
@@ -0,0 +1,153 @@
+		if ( WCS_ATT_Core_Compatibility::is_wc_version_gte_2_7() ) {
+			add_action( 'woocommerce_checkout_create_order_line_item', array( __CLASS__, 'save_subscription_scheme_meta' ), 10, 3 );
+		} else {
+			add_action( 'woocommerce_add_order_item_meta', array( __CLASS__, 'save_subscription_scheme_meta_legacy' ), 10, 2 );
+		}
+	}
+
+	/*
+	|--------------------------------------------------------------------------
+	| API
+	|--------------------------------------------------------------------------
+	*/
+
+	/**
+	 * Returns the key of the subscription scheme applied on the product when it was purchased.
+	 *
+	 * @param  array  $order_item
+	 * @return string|false|null
+	 */
+	public static function get_subscription_scheme( $order_item ) {
+		$scheme_key = null;
+		// Because who trusts WC 3.0 for back-compat?
+		if ( WCS_ATT_Core_Compatibility::is_wc_version_gte_2_7() ) {
+			if ( $order_item->meta_exists( '_wcsatt_scheme' ) ) {
+				$scheme_key = $order_item->get_meta( '_wcsatt_scheme', true );
+				$scheme_key = 0 === absint( $scheme_key ) ? false : strval( $scheme_key );
+			// Backwards compatibility with v1.
+			} elseif ( $order_item->meta_exists( '_wcsatt_scheme_id' ) ) {
+				$scheme_key = $order_item->get_meta( '_wcsatt_scheme_id', true );
+				$scheme_key = 0 === absint( $scheme_key ) ? false : strval( $scheme_key );
+			}
+		} else {
+			if ( isset( $order_item[ 'wcsatt_scheme' ] ) ) {
+				$scheme_key = 0 === absint( $order_item[ 'wcsatt_scheme' ] ) ? false : strval( $order_item[ 'wcsatt_scheme' ] );
+			// Backwards compatibility with v1.
+			} elseif ( isset( $order_item[ 'wcsatt_scheme_id' ] ) ) {
+				$scheme_key = 0 === absint( $order_item[ 'wcsatt_scheme_id' ] ) ? false : strval( $order_item[ 'wcsatt_scheme_id' ] );
+			}
+		}
+		return $scheme_key;
+	}
+
+	/*
+	|--------------------------------------------------------------------------
+	| Hooks
+	|--------------------------------------------------------------------------
+	*/
+
+	/**
+	 * Attempts to restore subscription data when creating a cart item using an order item as reference.
+	 *
+	 * @param  array     $cart_item
+	 * @param  array     $order_item
+	 * @param  WC_Order  $order
+	 * @return array
+	 */
+	public static function restore_cart_item_from_order_item( $cart_item, $order_item, $order ) {
+		if ( null !== ( $scheme_key = self::get_subscription_scheme( $order_item ) ) ) {
+			$cart_item[ 'wcsatt_data' ] = array(
+				'active_subscription_scheme' => $scheme_key
+			);
+		}
+		return $cart_item;
+	}
+
+	/**
+	 * Attempts to restore the subscription state of a product instantiated using an order item as reference.
+	 *
+	 * @param  WC_Product  $product
+	 * @param  array       $order_item
+	 * @return WC_Product
+	 */
+	public static function restore_product_from_order_item( $product, $order_item ) {
+		if ( null !== ( $scheme_key = self::get_subscription_scheme( $order_item ) ) ) {
+			WCS_ATT_Product::set_subscription_scheme( $product, $scheme_key );
```

Comment:
> If that's the behaviour, we'll probably want to prevent renewal/resubscribe to a product when the scheme has been deleted to avoid any confusion with attempting to resubscribe and the product no longer creating a subscription. Similar to the way you can't resubscribe to a subscription product that has since been deleted (but you can remain subscribed to it).

👍 I think that's the best approach. Opened https://github.com/Prospress/woocommerce-subscribe-all-the-things/issues/153